ext/php5/priority_sampling/priority_sampling.h

```diff
@@ -0,0 +1,15 @@
+#ifndef DDTRACE_PRIORITY_SAMPLING_H
+#define DDTRACE_PRIORITY_SAMPLING_H
+
+#include "../ddtrace.h"
+
+#define DDTRACE_PRIORITY_SAMPLING_UNKNOWN (1 << 30)
+#define DDTRACE_PRIORITY_SAMPLING_UNSET ((1 << 30) + 1)
+
+static const int PRIORITY_SAMPLING_USER_KEEP = 2;
+static const int PRIORITY_SAMPLING_USER_REJECT = -1;
```

Comment:
Why are `AUTO_REJECT` and `AUTO_KEEP` [not migrated](https://github.com/DataDog/dd-trace-php/blob/d3bc9f01302a94bc94465507fc61a01968da050d/src/DDTrace/Sampling/PrioritySampling.php#L10-L14) to the extension level as well?

Reply:
You mean, I shall define them, but not use them?

Reply:
This PR changes the default priority sampling value from `AUTO_KEEP` (1) to `USER_KEEP` (2). And `AUTO_REJECT` (0) to `USER_REJECT` (-1). This looks like a semantic change to me because this tells the Agent that all of the sampling decisions were explicitly set by the user even when they were not.

Looking at the existing implementation: The value generated from `dd_decide_on_sampling` was determined by the tracer so to mimic the exciting behavior, this would be `AUTO_KEEP`/`AUTO_REJECT`. And currently if a user explicitly set a `manual.keep`/`manual.drop` tag on a span, the sampling priority would be set to `USER_KEEP`/`USER_REJECT`.

So I'm curious if there was a reason on the Agent side of things to change this behavior in this PR. As far as I can tell, the original values are [still the ones we should be using here](https://github.com/DataDog/datadog-agent/blob/4b005758cdf6b5a63c9f3957ad8d82c17b9c558a/pkg/trace/sampler/sampler.go#L56-L71).

Reply:
Yes, this behavior is changing. Thanks for the link to the resource off thread, @bwoebi. 👍